stream/popularity_mapper.js

```diff
@@ -0,0 +1,215 @@
+  operator: { _score: 1000 },
+  fee: { _score: 2000 },
+
+  // closed and abandoned places
+  // https://wiki.openstreetmap.org/wiki/Key:disused:
+  disused: { _score: -100000 },
+  'disused:shop': { _score: -100000 },
+  'disused:amenity': { _score: -100000 },
+  'disused:railway': { _score: -100000 },
+  'disused:leisure': { _score: -100000 },
+  // https://wiki.openstreetmap.org/wiki/Key:abandoned:
+  abandoned: { _score: -100000 },
+  'abandoned:shop': { _score: -100000 },
+  'abandoned:amenity': { _score: -100000 },
+  'abandoned:railway': { _score: -100000 },
+  'abandoned:leisure': { _score: -100000 },
+
+  // types of public amenities
+  amenity: {
+    disused: { _score: -100000 },
+    college: { _score: 2000 },
+    library: { _score: 2000 },
+    school: { _score: 1000 },
+    university: { _score: 2000 },
+    bank: { _score: 1000 },
+    clinic: { _score: 1000 },
+    dentist: { _score: 1000 },
+    doctors: { _score: 1000 },
+    hospital: { _score: 2000 },
+    nursing_home: { _score: 1000 },
+    pharmacy: { _score: 1000 },
+    social_facility: { _score: 1000 },
+    veterinary: { _score: 1000 },
+    community_centre: { _score: 1000 },
+    music_venue: { _score: 1000 },
+    nightclub: { _score: 1000 },
+    planetarium: { _score: 1000 },
+    social_centre: { _score: 1000 },
+    theatre: { _score: 1000 },
+    courthouse: { _score: 1000 },
+    coworking_space: { _score: 1000 },
+    dojo: { _score: 1000 },
+    embassy: { _score: 1000 },
+    ferry_terminal: { _score: 2000 },
+    fire_station: { _score: 1000 },
+    marketplace: { _score: 1000 },
+    place_of_worship: { _score: 1000 },
+    police: { _score: 1000 },
+    post_office: { _score: 1000 },
+    prison: { _score: 1000 },
+    public_bath: { _score: 1000 },
+    townhall: { _score: 1000 }
+  },
+
+  // transportation
+  aerodrome: {
+    international: { _score: 10000 },
+    regional: { _score: 5000 }
+  },
+  iata: {
+    _score: 5000,
+    none: { _score: -5000 }
+  },
+  railway: {
+    station: { _score: 2000 },
+    subway_entrance: { _score: 2000 },
+  },
+  public_transport: {
+    station: { _score: 2000 }
+    // no scoring boost for `public_transport:*`, as many unimportant records have other tags
+  },
+
+  // contact information
+  website: { _score: 200 },
+  phone: { _score: 200 },
+  'contact:website': { _score: 200 },
+  'contact:email': { _score: 200 },
+  'contact:phone': { _score: 200 },
+  'contact:fax': { _score: 200 },
+
+  // social media
+  'contact:foursquare': { _score: 200 },
+  'contact:facebook': { _score: 200 },
+  'contact:linkedin': { _score: 200 },
+  'contact:instagram': { _score: 200 },
+  'contact:skype': { _score: 200 },
+  'contact:flickr': { _score: 200 },
+  'contact:youtube': { _score: 200 },
+};
+
+module.exports = function(){
+
+  return through.obj(( doc, enc, next ) => {
+
+    try {
+
+      // only map venues
+      if( doc.getLayer() !== 'venue' ){
+        return next(null, doc);
+      }
```

Comment:
I was thinking we could remove this. Famous addresses such as for the [Sydney Opera House](https://pelias.github.io/compare/#/v1/search?text=2+Macquarie+Street%2C+Sydney%2C+AUS&debug=0) may not always appear first

Reply:
On second thought let's definitely either solve this separately, or at least enable popularity for addresses in a separate PR.